File: backend/app/schemas/research.py
"""研究 API 出入参。详情/摘要直接映射 ResearchTask；追加复用 state.schema 子结构。"""
from datetime import datetime

from pydantic import BaseModel


class CreateResearchTaskRequest(BaseModel):
    topic: str
    session_id: int | None = None


class AppendRequest(BaseModel):
    field: str  # facts/data_points/assumptions/charts/drafts/reviews
    items: list[dict]


class ResearchTaskSummary(BaseModel):
    id: int
    topic: str
    status: str
    version: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class ResearchTaskDetail(ResearchTaskSummary):
    outline: list
    assumptions: list
    facts: list
    data_points: list
    charts: list
    drafts: list
    final: dict | None
    reviews: list


class TrajectoryResponse(BaseModel):
    id: int
    task_id: int
    topic: str
    summary: str
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/types/research.ts
export interface Provenance {
  agent: string;
  source: string | null;
  created_at: string;
}

export interface ResearchTaskSummary {
  id: number;
  topic: string;
  status: string;
  version: number;
  created_at: string;
  updated_at: string;
}

export interface ResearchTaskDetail extends ResearchTaskSummary {
  outline: Array<{ id: string; title: string; points: string[]; provenance: Provenance }>;
  assumptions: Array<Record<string, unknown>>;
  facts: Array<{ id: string; content: string; provenance: Provenance }>;
  data_points: Array<Record<string, unknown>>;
  charts: Array<Record<string, unknown>>;
  drafts: Array<Record<string, unknown>>;
  final: Record<string, unknown> | null;
  reviews: Array<Record<string, unknown>>;
}

export interface CreateResearchTaskPayload {
  topic: string;
  session_id?: number;
}
